kinto/core/authorization.py

```diff
@@ -196,7 +196,12 @@ def get_permission_object_id(self, request, object_id=None):
                                                 self.resource_name,
                                                 **matchdict)
                 if object_id == '*':
-                    object_uri = object_uri.replace('%2A', '*')
+                    # Express the object_uri pattern with precision
+                    # XXX: we could get the model.id_generator pattern, but
+                    # since the backend replace * by .* we can't.
+                    # See Kinto/kinto#970
+                    id_pattern = "[a-zA-Z0-9_-]+"
+                    object_uri = object_uri.replace('%2A', id_pattern)
```

Comment:
This is fix. But since the history resource was relying on the bug to work, I had to do some changes regarding the history entries (eg. explicit declare that the permissions inherit from bucket)